### DISCUSSION TITLE
Expanded use of LangChain features - tools, etc.?

### DISCUSSION THREAD

**Discussion by radman-x** [AntonOsika/gpt-engineer#602] [Ideas]
Thanx for creating such a good ai dev aide. I know GE is using LangChain for LLM abstraction, which is great. Can you expand GE functionality to add more LangChain features like external tools, etc? Eg., to allow web searching, GitHub repo access, chat with private documents, etc.

Is there any interest in the GE community for this?

**Comment by AntonOsika** [comment]
This sounds interesting! PoC:s welcome.

Would be great to start with an important problem to solve for users and then see what tools we need for this.

**Comment by radman-x** [comment]
One good use case is to allow GE itself code GE issues, using its GitHub repo’s issues as the initial prompt. Eg, it would read #682 form the repo, code up the changes for implementing the “code review” and self healing step, create a PR and commit all the changes to the PR for human review.

That is just one uc. It can also help with use cases listed in #166, etc.
